From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Fri, 21 Jul 2017 19:59:33 +0200
Subject: [PATCH] lib.sh: Use host-passthrough when spawning VMs

virsh defaults to using "host-model" instead of "host-passthrough",
which not only might lead to a small performance hit on x86, it is
also causing compatibility issues with libvirt 3.x.

Address this by explicitly requesting "host-passthrough" as the
CPU model for virt-install.

Signed-off-by: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
---
 mcp/scripts/lib.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mcp/scripts/lib.sh b/mcp/scripts/lib.sh
index b75e47e..55df750 100644
--- a/mcp/scripts/lib.sh
+++ b/mcp/scripts/lib.sh
@@ -67,7 +67,7 @@ create_vms() {

   # create vms with specified options
   for node in "${vnodes[@]}"; do
-    virt-install --name ${node} --ram ${vnodes_ram[$node]} --vcpus ${vnodes_vcpus[$node]} --cpu host --accelerate \
+    virt-install --name ${node} --ram ${vnodes_ram[$node]} --vcpus ${vnodes_vcpus[$node]} --cpu host-passthrough --accelerate \
     --network network:pxe,model=virtio \
     --network network:mgmt,model=virtio \
     --network network:internal,model=virtio \
